### PR TITLE
Add "Monthly Charges" Card

### LINKED ISSUE #2379: Page Layouts - Four column dashboard "Monthly Charges" card
Implement https://www.figma.com/file/4rdBkUlHd5MCVq3hvUOXHd/Page-Layouts?node-id=2677%3A22072

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-2390--keen-mayer-a86c8b.netlify.app/templates/dashboards/#four-column-dashboard)

#### What does this PR do?

Implements "Monthly Charges" Card. I have just used default graph colors because chart doesn't support a gradient. Additionally, I kept the gap between the title and measure at "24px" as opposed to "48px" in designs because the gap seemed unnecessarily large.

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [x] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [x] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### What are the relevant issues?
Closes #2379 
#### Is this change backwards compatible or is it a breaking change?
backwards compatible.

### CODE REVIEW

File: aries-site/src/examples/templates/dashboards/content/chartCards/ChargesByMonth.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Grid, DataChart, Text } from 'grommet';
+import { ChartCard, Measure } from '../../components';
+import { defaultWindow, formatCurrency, REPORT_WINDOW_MAP } from './utils';
+
+const MOCK_DATA = require('../../../../../data/mockData/consumption.json');
+
+export const ChargesByMonth = ({ period }) => {
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/ChargesByMonth.js:9]: This seems like a 95% overlap of the CostByMonth. I am wondering if we should think about a more reusable approach here. Throwing out some ideas:

- Should we just consume the existing component from within the 4-col dashboard?
- Should we create a BarChartCard, which could be called by both ChargesByMonth and CostByMonth?
- Should we move the 'Set reporting window' and 'Filter and assemble data set' useEffects into their own reusable hooks (and migrate them to memos)?

Maybe we should make MonthlySpend the reusable component + make the hooks reusable?

File: aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  REPORT_WINDOW_MAP,
+} from '.';
+
+export const useCost = values => {
+  const [meanCost, setMeanCost] = useState(null);
+  const [lastMonthCost, setLastMonthCost] = useState(null);
+  const [projectedCost, setProjectedCost] = useState(null);
+
+  // Calculate cost stats
+  useEffect(() => {
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js:16]: Any reason that this cannot be executed as a useMemo here?

File: aries-site/src/examples/templates/dashboards/content/chartCards/CostByMonth.js [hunk 3]
```diff
@@ -1,117 +1,25 @@
...
-          dateMap.get(
-            `${nextMonth.getMonth() +
-              1}/${nextMonth.getDate()}/${nextMonth.getFullYear()}`,
-          ) +
-        b;
-      setProjectedCost(costNextMonth.toFixed(2));
-    }
-  }, [values]);
+  // ensure component is mounted before trying to access
+  // browser's Navigator
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/CostByMonth.js:21]: 👍

File: aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    }
+  }, [values]);
+
+  return { meanCost, lastMonthCost, projectedCost };
+};
+
+export const useDataSet = (consumptionData, reportWindow) => {
+  const values = useMemo(() => {
+    if (consumptionData) {
+      const nextValues = [];
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js:70]: This plus following suggestion:

```suggestion
    const nextValues = [];
    if (consumptionData) {
```

File: aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+              parseFloat(nextValues[index].value) + parseFloat(value)
+            ).toFixed(2);
+            nextValues[index].value = Number(nextValue);
+          }
+        }
+      });
+      nextValues.sort((a, b) => b.key - a.key);
+      return nextValues;
+    }
+    return null;
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js:97]: This plus previous suggesion:

```suggestion
    }
    return nextValues;
```

File: aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+          ) +
+        b;
+      setProjectedCost(costNextMonth.toFixed(2));
+    }
+  }, [values]);
+
+  return { meanCost, lastMonthCost, projectedCost };
+};
+
+export const useDataSet = (consumptionData, reportWindow) => {
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js:67]: Naming + comment

```suggestion
// Filter and assemble data set. Limit consumptionData to datapoints within
// the reportWindow range. Create data set which sums spend by month.
export const useSpendByMonth = (consumptionData, reportWindow) => {
```

File: aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  return reportWindow;
+};
+
+// ensure component is mounted before trying to access
+// browser's Navigator
+export const useMounted = () => {
+  const [mounted, setMounted] = useState('unmounted');
+  useEffect(() => {
+    setMounted('mounted');
+    return () => setMounted('unmounted');
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js:125]: 👍

File: aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+    }));
+  }, [period]);
+
+  return reportWindow;
+};
+
+// ensure component is mounted before trying to access
+// browser's Navigator
+export const useMounted = () => {
+  const [mounted, setMounted] = useState('unmounted');
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js:122]: Any reason not to treat as boolean?

File: aries-site/src/examples/templates/dashboards/content/chartCards/CostByMonth.js [hunk 9]
```diff
@@ -132,29 +40,33 @@ export const CostByMonth = ({ period }) => {
...
-            gridArea="projection"
-            name={{
-              label: { label: 'Projected, Next Month', size: 'medium' },
-            }}
-            value={{
-              value: formatCurrency(projectedCost, Navigator.language),
-              size: 'xlarge',
-            }}
-          />
+          {mounted === 'mounted' && (
```
Comment (halocline) [inline aries-site/src/examples/templates/dashboards/content/chartCards/CostByMonth.js:43]: Convert to boolean?

**Review by halocline — COMMENTED** [review]
This is looking beautiful. I'm glad we went this direction. So much cleaner and should be much easier for folks to follow.

Just added a few questions / suggestions for polishing.

File: aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js (hunk 8 above)
Comment (taysea) [reply aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js:122]: Simplified to boolean.

File: aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js (hunk 2 above)
Comment (taysea) [reply aries-site/src/examples/templates/dashboards/content/chartCards/utils/hooks.js]: Thanks, have updated to a memo.

**Review by halocline — APPROVED** [review]

**Pull request merged**